cmd/scollector/main.go

```diff
@@ -159,81 +149,63 @@ func main() {
 	for _, m := range mains {
 		m()
 	}
-	readConf()
-	if *flagTags != "" {
-		var err error
-		collectors.AddTags, err = opentsdb.ParseTags(*flagTags)
-		if err != nil {
-			slog.Fatalf("failed to parse additional tags %v: %v", *flagTags, err)
-		}
-		if collectors.AddTags["host"] != "" {
-			slog.Fatalf("host not supported in custom tags, use -hostname instead")
-		}
+	conf := readConf()
+	if !conf.Tags.Valid() {
+		slog.Fatalf("invalid tags: %v", conf.Tags)
+	} else if conf.Tags["host"] != "" {
+		slog.Fatalf("host not supported in custom tags, use Hostname instead")
 	}
-	util.FullHostname = *flagFullHost
+	collectors.AddTags = conf.Tags
+	util.FullHostname = conf.FullHost
 	util.Set()
-	if *flagHostname != "" {
-		util.Hostname = *flagHostname
-		if err := collect.SetHostname(*flagHostname); err != nil {
+	if conf.Hostname != "" {
+		util.Hostname = conf.Hostname
+		if err := collect.SetHostname(conf.Hostname); err != nil {
 			slog.Fatal(err)
 		}
 	}
-	if *flagColDir != "" {
-		collectors.InitPrograms(*flagColDir)
+	if conf.ColDir != "" {
+		collectors.InitPrograms(conf.ColDir)
 	}
-	if *flagSNMP != "" {
-		for _, s := range strings.Split(*flagSNMP, ",") {
-			sp := strings.Split(s, "@")
-			if len(sp) != 2 {
-				slog.Fatal("invalid snmp string:", *flagSNMP)
-			}
-			collectors.SNMPIfaces(sp[0], sp[1])
-			collectors.SNMPCisco(sp[0], sp[1])
+	for _, h := range conf.HAProxy {
+		for _, i := range h.Instances {
+			collectors.HAProxy(h.User, h.Password, i.Tier, i.URL)
 		}
 	}
-	if *flagICMP != "" {
-		for _, s := range strings.Split(*flagICMP, ",") {
-			collectors.ICMP(s)
+	for _, s := range conf.SNMP {
```

Comment:
Can individual collectors do this kind of configuring on their own on init? Main would be cleaner if fewer collector-specific things where there?